### ISSUE TITLE
fluent-bit fails to forward logs with 'no upstream connections available to <endpoint>'

### ISSUE THREAD

**Issue by shreesha21** [fluent/fluent-bit#6780]
## Bug Report

**Describe the bug**
We wanted to ship some MinIO logs to an internal Log Insight endpoint and we have configured the fluent-bit operator to do so.
Even though the operator logs shows HTTP 200 in the beginning (just a couple of times) but later complains with `no upstream connections available to <lint endpoint>.

- Example log message if applicable:

(Fluent-bit log snipped with debug enabled):
```
[2023/02/01 03:26:24] [debug] [out flush] cb_destroy coro_id=0
[2023/02/01 03:26:24] [debug] [task] destroy task=0x7fe82c03d3e0 (task_id=0)
[2023/02/01 03:26:33] [debug] [input chunk] update output instances with new chunk size diff=514
[2023/02/01 03:26:33] [debug] [input chunk] update output instances with new chunk size diff=3009
[2023/02/01 03:26:35] [debug] [input chunk] update output instances with new chunk size diff=514
[2023/02/01 03:26:35] [debug] [input chunk] update output instances with new chunk size diff=2996
[2023/02/01 03:26:35] [debug] [task] created task=0x7fe82c0c0580 id=0 OK
[2023/02/01 03:26:35] [debug] [output:http:http.0] task_id=0 assigned to thread #1
[2023/02/01 03:26:36] [debug] [http_client] not using http_proxy for header
[2023/02/01 03:26:36] [ info] [output:http:http.0] data.mgmt.cloud.vmware.com:443, HTTP status=200
{}
[2023/02/01 03:26:36] [debug] [out flush] cb_destroy coro_id=0
[2023/02/01 03:26:36] [debug] [task] destroy task=0x7fe82c0c0580 (task_id=0)
[2023/02/01 03:29:20] [debug] [input chunk] update output instances with new chunk size diff=672
[2023/02/01 03:29:20] [debug] [input chunk] update output instances with new chunk size diff=558
[2023/02/01 03:29:20] [debug] [task] created task=0x7fe82c03c630 id=0 OK
[2023/02/01 03:29:20] [debug] [output:http:http.0] task_id=0 assigned to thread #0
[2023/02/01 03:29:20] [ warn] [net] getaddrinfo(host='data.mgmt.cloud.vmware.com', err=11): Could not contact DNS servers
[2023/02/01 03:29:20] [debug] [upstream] connection #-1 failed to data.mgmt.cloud.vmware.com:443
[2023/02/01 03:29:20] [error] [output:http:http.0] no upstream connections available to data.mgmt.cloud.vmware.com:443
[2023/02/01 03:29:20] [debug] [out flush] cb_destroy coro_id=1
[2023/02/01 03:29:20] [debug] [retry] new retry created for task_id=0 attempts=1
[2023/02/01 03:29:20] [ warn] [engine] failed to flush chunk '1-1675222160.756791381.flb', retry in 11 seconds: task_id=0, input=http.0 > output=http.0 (out_id=0)
```

- Steps to reproduce the problem:
* Deploy a kind cluster
* Install MinIO tenant
* Install Fluent-bit
* Configure Fluent bit config map to forward logs to a Log Insight endpoint

**Expected behavior**
We expected to have these MinIO operator logs successfully get forwarded to the Log Insight endpoint where in we can actually view them on the Grafana UI.

**Your Environment**
* Version used:
Fluent-bit operator version: v2.0.8
Minio Version: RELEASE.2023-01-25T00-19-54Z 

* Configuration:
```
❯ k -n infra-0f9740-rs-lint-wdc get cm minio-0f9740-rs-lint-wdc-log -oyaml
apiVersion: v1
data:
  fluentbit.conf: |-
    # fluentbit.conf
    [SERVICE]
        log_level debug
        flush 3

    # Listen on the default IP/Port - 0.0.0.0:9880
    [INPUT]
        name http
        successful_response_code 200

    # We want to add the namespace name to know from which MinIO Cluster this message came from
    [FILTER]
        Name record_modifier
        Match *
        Record minio_namespace infra-0f9740-rs-lint-wdc

    # This is an example output snippet to use with vRealize Log Insight Cloud
    [OUTPUT]
        Name     http
        Match    *
        tls      On
        Host     data.mgmt.cloud.vmware.com
        Port     443
        URI      /le-mans/v1/streams/ingestion-pipeline-stream
        Format   json
        Header   Authorization Bearer <AuthToken>
kind: ConfigMap
metadata:
  creationTimestamp: "2023-01-31T21:01:22Z"
  name: minio-0f9740-rs-lint-wdc-log
  namespace: infra-0f9740-rs-lint-wdc
  resourceVersion: "136797747"
  selfLink: /api/v1/namespaces/infra-0f9740-rs-lint-wdc/configmaps/minio-0f9740-rs-lint-wdc-log
  uid: 083c123f-597f-472e-aba5-39ef7fa927d3
```
* Environment name and version (e.g. Kubernetes? What version?):
Kubernetes Version: 
```
❯ kubectl version
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.2", GitCommit:"5835544ca568b757a8ecae5c153f317e5736700e", GitTreeState:"clean", BuildDate:"2022-09-21T14:25:45Z", GoVersion:"go1.19.1", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.5+vmware.wcp.2", GitCommit:"0f23d9e39892fef2a7e9da16d6b517d579a3164d", GitTreeState:"clean", BuildDate:"2022-05-26T18:50:09Z", GoVersion:"go1.17.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by lecaros** [comment]
Is this error repeatable and consistent?
which image of fluent-bit are you using?
is there any network component in the middle that could be dropping the connection?

**Comment by shreesha21** [comment]
Yes, it is consistently reproducible.
Fluent-bit image: fluent/fluent-bit:2.0.8

> is there any network component in the middle that could be dropping the connection?
I don't think so. How do you suggest verify this?

Further, as I cannot get into the shell with this released version, I used the 2.0.8-debug released image to drop into the shell and ran this command and it gave me a successful response.

```
root@minio-0f9740-rs-lint-wdc-log-dc7bf5468-8rv89:/# curl -XPOST -H "Authorization: Bearer XXXXX" https://data.mgmt.cloud.vmware.com/le-mans/v1/streams/ingestion-pipeline-stream -v
*   Trying 52.32.98.22:443...
* Connected to data.mgmt.cloud.vmware.com (52.32.98.22) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=api.mgmt.cloud.vmware.com
*  start date: Sep 13 00:00:00 2022 GMT
*  expire date: Oct 11 23:59:59 2023 GMT
*  subjectAltName: host "data.mgmt.cloud.vmware.com" matched cert's "data.mgmt.cloud.vmware.com"
*  issuer: C=US; O=Amazon; OU=Server CA 1B; CN=Amazon
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x560a0939f2c0)
> POST /le-mans/v1/streams/ingestion-pipeline-stream HTTP/2
> Host: data.mgmt.cloud.vmware.com
> user-agent: curl/7.74.0
> accept: */*
> authorization: Bearer XXXXX
>
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
< HTTP/2 200
< date: Fri, 03 Feb 2023 17:47:13 GMT
< content-type: application/json
< content-length: 2
<
* Connection #0 to host data.mgmt.cloud.vmware.com left intact
```

This tells me that it can successfully resolve and reach the endpoint but I don't know why it is failing when forwarding the logs.
